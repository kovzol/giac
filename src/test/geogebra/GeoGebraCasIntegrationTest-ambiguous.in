[[[ggbevalarg:=[[[ggbelarg0:=when ( type((((((((((((8)*((x)^(2)))*(sqrt(10)))+((12)*((x)^(2))))-(((32)*(x))*(sqrt(10))))-(((16)*(x))*(y)))-((24)*(x)))+(((8)*(sqrt(10)))*((y)^(2))))-(((24)*(sqrt(10)))*(y)))+((32)*(sqrt(10))))+((24)*((y)^(2))))-((40)*(y))=0) == DOM_SYMBOLIC && type(y) == DOM_SYMBOLIC , (assume(y),solve((((((((((((8)*((x)^(2)))*(sqrt(10)))+((12)*((x)^(2))))-(((32)*(x))*(sqrt(10))))-(((16)*(x))*(y)))-((24)*(x)))+(((8)*(sqrt(10)))*((y)^(2))))-(((24)*(sqrt(10)))*(y)))+((32)*(sqrt(10))))+((24)*((y)^(2))))-((40)*(y))=0,x))[size(assume(y),solve((((((((((((8)*((x)^(2)))*(sqrt(10)))+((12)*((x)^(2))))-(((32)*(x))*(sqrt(10))))-(((16)*(x))*(y)))-((24)*(x)))+(((8)*(sqrt(10)))*((y)^(2))))-(((24)*(sqrt(10)))*(y)))+((32)*(sqrt(10))))+((24)*((y)^(2))))-((40)*(y))=0,x))-1] , when ( type((((((((((((8)*((x)^(2)))*(sqrt(10)))+((12)*((x)^(2))))-(((32)*(x))*(sqrt(10))))-(((16)*(x))*(y)))-((24)*(x)))+(((8)*(sqrt(10)))*((y)^(2))))-(((24)*(sqrt(10)))*(y)))+((32)*(sqrt(10))))+((24)*((y)^(2))))-((40)*(y))=0) == DOM_IDENT && type(y) == DOM_SYMBOLIC && (((((((((((8)*((x)^(2)))*(sqrt(10)))+((12)*((x)^(2))))-(((32)*(x))*(sqrt(10))))-(((16)*(x))*(y)))-((24)*(x)))+(((8)*(sqrt(10)))*((y)^(2))))-(((24)*(sqrt(10)))*(y)))+((32)*(sqrt(10))))+((24)*((y)^(2))))-((40)*(y))=0 == 'x', (assume(y),solve((((((((((((8)*((x)^(2)))*(sqrt(10)))+((12)*((x)^(2))))-(((32)*(x))*(sqrt(10))))-(((16)*(x))*(y)))-((24)*(x)))+(((8)*(sqrt(10)))*((y)^(2))))-(((24)*(sqrt(10)))*(y)))+((32)*(sqrt(10))))+((24)*((y)^(2))))-((40)*(y))=0=0,x))[size(assume(y),solve((((((((((((8)*((x)^(2)))*(sqrt(10)))+((12)*((x)^(2))))-(((32)*(x))*(sqrt(10))))-(((16)*(x))*(y)))-((24)*(x)))+(((8)*(sqrt(10)))*((y)^(2))))-(((24)*(sqrt(10)))*(y)))+((32)*(sqrt(10))))+((24)*((y)^(2))))-((40)*(y))=0=0,x))-1] ,when ( size(y) == 1,flatten1(ggbsort(normal([op(solve((((((((((((8)*((x)^(2)))*(sqrt(10)))+((12)*((x)^(2))))-(((32)*(x))*(sqrt(10))))-(((16)*(x))*(y)))-((24)*(x)))+(((8)*(sqrt(10)))*((y)^(2))))-(((24)*(sqrt(10)))*(y)))+((32)*(sqrt(10))))+((24)*((y)^(2))))-((40)*(y))=0,y))]))),ggbsort(normal([op(solve((((((((((((8)*((x)^(2)))*(sqrt(10)))+((12)*((x)^(2))))-(((32)*(x))*(sqrt(10))))-(((16)*(x))*(y)))-((24)*(x)))+(((8)*(sqrt(10)))*((y)^(2))))-(((24)*(sqrt(10)))*(y)))+((32)*(sqrt(10))))+((24)*((y)^(2))))-((40)*(y))=0,y))])) ) ) )], [ggbelarg1:=1]],when(type(ggbelarg0)==DOM_LIST,(ggbelarg0)[ggbelarg1-1],when(ggbelarg1>0,(ggbelarg0)[ggbelarg1],?))][1]],[ggbevalans:=regroup(normal(ggbevalarg))],[ggbevalans2:=regroup(ggbevalarg)]], when(length(""+ggbevalans)<=length(""+ggbevalans2),ggbevalans,ggbevalans2)][1]
when ( type((((ggbtmpvara)*((x)^(2)))+((ggbtmpvarb)*(x)))+(ggbtmpvarc)) == DOM_SYMBOLIC && type(x) == DOM_SYMBOLIC , (assume(x),solve((((ggbtmpvara)*((x)^(2)))+((ggbtmpvarb)*(x)))+(ggbtmpvarc),x))[size(assume(x),solve((((ggbtmpvara)*((x)^(2)))+((ggbtmpvarb)*(x)))+(ggbtmpvarc),x))-1] , when ( type((((ggbtmpvara)*((x)^(2)))+((ggbtmpvarb)*(x)))+(ggbtmpvarc)) == DOM_IDENT && type(x) == DOM_SYMBOLIC && (((ggbtmpvara)*((x)^(2)))+((ggbtmpvarb)*(x)))+(ggbtmpvarc) == 'x', (assume(x),solve((((ggbtmpvara)*((x)^(2)))+((ggbtmpvarb)*(x)))+(ggbtmpvarc)=0,x))[size(assume(x),solve((((ggbtmpvara)*((x)^(2)))+((ggbtmpvarb)*(x)))+(ggbtmpvarc)=0,x))-1] ,when ( size(x) == 1,flatten1(ggbsort(normal([op(solve((((ggbtmpvara)*((x)^(2)))+((ggbtmpvarb)*(x)))+(ggbtmpvarc),x))]))),ggbsort(normal([op(solve((((ggbtmpvara)*((x)^(2)))+((ggbtmpvarb)*(x)))+(ggbtmpvarc),x))])) ) ) )
[[[ggbtanarg0:=ggbtmpvara],[ggbtanarg1:=sqrt((1)-((x)^(2)))],[ggbtanvar:=when(size(lname(ggbtanarg1) intersect [x]) == 0,lname(ggbtanarg1)[0],x)]],when((ggbtmpvara)[0]=='pnt',when((ggbtanarg1)[0]=='=',equation(tangent(ggbtanarg1,ggbtanarg0)),y=normal(subst(diff(ggbtanarg1,ggbtanvar),ggbtanvar=real(ggbtanarg0[1]))*(x-real(ggbtanarg0[1]))+subst(ggbtanarg1,ggbtanvar=real(ggbtmpvara[1])))),y=normal(subst(diff(ggbtanarg1,ggbtanvar),ggbtanvar=ggbtanarg0)*(x-(ggbtanarg0))+subst(ggbtanarg1,ggbtanvar=ggbtanarg0)))][1]
