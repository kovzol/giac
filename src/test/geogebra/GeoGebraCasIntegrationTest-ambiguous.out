1>> caseval("init geogebra")
1<< "geogebra mode on"
2>> fractionalPart(x):=sign(x)*(abs(x)-floor(abs(x)))
2<<  (x)->sign(x)*(abs(x)-floor(abs(x)))
3>> is3dpoint(x):=when(size(x[1])==3 && subtype(x[1])==20,true,false)
3<<  (x)->when((((size(x[1]))==3)) && (((subtype(x[1]))==20)),true,false)
4>> xcoord(a):=when(type(a)==DOM_IDENT,xcoordsymb(a),when(a[0]=='pnt',when(is3dpoint(a),a[1][0],real(a[1])),when(a[0]=='=',coeff(a[1]-a[2],x,1),a[0])))
4<<  (a)->when(((type(a))==identifier),xcoordsymb(a),when(((a[0])=='pnt'),when(is3dpoint(a),(a[1])[0],real(a[1])),when(((a[0])=='='),coeff(a[1]-(a[2]),x,1),a[0])))
5>> ycoord(a):=when(type(a)==DOM_IDENT,ycoordsymb(a),when(a[0]=='pnt',when(is3dpoint(a),a[1][1],im(a[1])),when(a[0]=='=',coeff(a[1]-a[2],y,1),a[1])))
5<<  (a)->when(((type(a))==identifier),ycoordsymb(a),when(((a[0])=='pnt'),when(is3dpoint(a),(a[1])[1],im(a[1])),when(((a[0])=='='),coeff(a[1]-(a[2]),y,1),a[1])))
6>> zcoord(a):=when(type(a)==DOM_IDENT,zcoordsymb(a),when(a[0]=='pnt',when(is3dpoint(a),a[1][2],0),when(length(a)<3 && a[0] != '=',0,when(a[0]=='=',coeff(a[1]-a[2],z,1),a[2]))))
6<<  (a)->when(((type(a))==identifier),zcoordsymb(a),when(((a[0])=='pnt'),when(is3dpoint(a),(a[1])[2],0),when(((length(a))<3) && ((a[0])<>'='),0,when(((a[0])=='='),coeff(a[1]-(a[2]),z,1),a[2]))))
7>> degasin(x):=normal(asin(x)/pi*180)*unicode0176u
7<<  (x)->normal(asin(x)/pi*180)*unicode0176u
8>> degacos(x):=normal(acos(x)/pi*180)*unicode0176u
8<<  (x)->normal(acos(x)/pi*180)*unicode0176u
9>> degatan(x):=normal(atan(x)/pi*180)*unicode0176u
9<<  (x)->normal(atan(x)/pi*180)*unicode0176u
10>> degatan2(y,x):=normal(arg(x+i*y)/pi*180)*unicode0176u
10<<  (y,x)->normal(arg(x+ί*y)/pi*180)*unicode0176u
11>> ggbabs(x):=when(x[0]=='pnt' || (type(x)==DOM_LIST && subtype(x)==27),l2norm(x),abs(x))
11<<  (x)->when((((x[0])=='pnt')) || ((((type(x))==vector)) && (((subtype(x))==27))),l2norm(x),abs(x))
12>> ggb_is_zero(x):=when(x==0,true,when(type(x)=='DOM_LIST',max(flatten({x,0}))==min(flatten({x,0}))&&min(flatten({x,0}))==0,when(x[0]=='=',lhs(x)==0&&rhs(x)==0,false)))
12<<  (x)->when((x==0),true,when(((type(x))==vector),(((max(flatten({x,0})))==(min(flatten({x,0}))))) && (((min(flatten({x,0})))==0)),when(((x[0])=='='),(((lhs(x))==0)) && (((rhs(x))==0)),false)))
13>> caseval(regroup([[[ggbevalarg:=[[[ggbelarg0:=when ( type((((((((((((8)*((x)^(2)))*(sqrt(10)))+((12)*((x)^(2))))-(((32)*(x))*(sqrt(10))))-(((16)*(x))*(y)))-((24)*(x)))+(((8)*(sqrt(10)))*((y)^(2))))-(((24)*(sqrt(10)))*(y)))+((32)*(sqrt(10))))+((24)*((y)^(2))))-((40)*(y))=0) == DOM_SYMBOLIC && type(y) == DOM_SYMBOLIC , (assume(y),solve((((((((((((8)*((x)^(2)))*(sqrt(10)))+((12)*((x)^(2))))-(((32)*(x))*(sqrt(10))))-(((16)*(x))*(y)))-((24)*(x)))+(((8)*(sqrt(10)))*((y)^(2))))-(((24)*(sqrt(10)))*(y)))+((32)*(sqrt(10))))+((24)*((y)^(2))))-((40)*(y))=0,x))[size(assume(y),solve((((((((((((8)*((x)^(2)))*(sqrt(10)))+((12)*((x)^(2))))-(((32)*(x))*(sqrt(10))))-(((16)*(x))*(y)))-((24)*(x)))+(((8)*(sqrt(10)))*((y)^(2))))-(((24)*(sqrt(10)))*(y)))+((32)*(sqrt(10))))+((24)*((y)^(2))))-((40)*(y))=0,x))-1] , when ( type((((((((((((8)*((x)^(2)))*(sqrt(10)))+((12)*((x)^(2))))-(((32)*(x))*(sqrt(10))))-(((16)*(x))*(y)))-((24)*(x)))+(((8)*(sqrt(10)))*((y)^(2))))-(((24)*(sqrt(10)))*(y)))+((32)*(sqrt(10))))+((24)*((y)^(2))))-((40)*(y))=0) == DOM_IDENT && type(y) == DOM_SYMBOLIC && (((((((((((8)*((x)^(2)))*(sqrt(10)))+((12)*((x)^(2))))-(((32)*(x))*(sqrt(10))))-(((16)*(x))*(y)))-((24)*(x)))+(((8)*(sqrt(10)))*((y)^(2))))-(((24)*(sqrt(10)))*(y)))+((32)*(sqrt(10))))+((24)*((y)^(2))))-((40)*(y))=0 == 'x', (assume(y),solve((((((((((((8)*((x)^(2)))*(sqrt(10)))+((12)*((x)^(2))))-(((32)*(x))*(sqrt(10))))-(((16)*(x))*(y)))-((24)*(x)))+(((8)*(sqrt(10)))*((y)^(2))))-(((24)*(sqrt(10)))*(y)))+((32)*(sqrt(10))))+((24)*((y)^(2))))-((40)*(y))=0=0,x))[size(assume(y),solve((((((((((((8)*((x)^(2)))*(sqrt(10)))+((12)*((x)^(2))))-(((32)*(x))*(sqrt(10))))-(((16)*(x))*(y)))-((24)*(x)))+(((8)*(sqrt(10)))*((y)^(2))))-(((24)*(sqrt(10)))*(y)))+((32)*(sqrt(10))))+((24)*((y)^(2))))-((40)*(y))=0=0,x))-1] ,when ( size(y) == 1,flatten1(ggbsort(normal([op(solve((((((((((((8)*((x)^(2)))*(sqrt(10)))+((12)*((x)^(2))))-(((32)*(x))*(sqrt(10))))-(((16)*(x))*(y)))-((24)*(x)))+(((8)*(sqrt(10)))*((y)^(2))))-(((24)*(sqrt(10)))*(y)))+((32)*(sqrt(10))))+((24)*((y)^(2))))-((40)*(y))=0,y))]))),ggbsort(normal([op(solve((((((((((((8)*((x)^(2)))*(sqrt(10)))+((12)*((x)^(2))))-(((32)*(x))*(sqrt(10))))-(((16)*(x))*(y)))-((24)*(x)))+(((8)*(sqrt(10)))*((y)^(2))))-(((24)*(sqrt(10)))*(y)))+((32)*(sqrt(10))))+((24)*((y)^(2))))-((40)*(y))=0,y))])) ) ) )], [ggbelarg1:=1]],when(type(ggbelarg0)==DOM_LIST,(ggbelarg0)[ggbelarg1-1],when(ggbelarg1>0,(ggbelarg0)[ggbelarg1],?))][1]],[ggbevalans:=regroup(normal(ggbevalarg))],[ggbevalans2:=regroup(ggbevalarg)]], when(length(""+ggbevalans)<=length(""+ggbevalans2),ggbevalans,ggbevalans2)][1]))
13<< y=((2*x+3*√10-3*√(x^2*(-2*√10-6)+x*(8*√10+24)-2*√10-5)+5)/(2*√10+6))
14>> caseval(regroup(when ( type((((ggbtmpvara)*((x)^(2)))+((ggbtmpvarb)*(x)))+(ggbtmpvarc)) == DOM_SYMBOLIC && type(x) == DOM_SYMBOLIC , (assume(x),solve((((ggbtmpvara)*((x)^(2)))+((ggbtmpvarb)*(x)))+(ggbtmpvarc),x))[size(assume(x),solve((((ggbtmpvara)*((x)^(2)))+((ggbtmpvarb)*(x)))+(ggbtmpvarc),x))-1] , when ( type((((ggbtmpvara)*((x)^(2)))+((ggbtmpvarb)*(x)))+(ggbtmpvarc)) == DOM_IDENT && type(x) == DOM_SYMBOLIC && (((ggbtmpvara)*((x)^(2)))+((ggbtmpvarb)*(x)))+(ggbtmpvarc) == 'x', (assume(x),solve((((ggbtmpvara)*((x)^(2)))+((ggbtmpvarb)*(x)))+(ggbtmpvarc)=0,x))[size(assume(x),solve((((ggbtmpvara)*((x)^(2)))+((ggbtmpvarb)*(x)))+(ggbtmpvarc)=0,x))-1] ,when ( size(x) == 1,flatten1(ggbsort(normal([op(solve((((ggbtmpvara)*((x)^(2)))+((ggbtmpvarb)*(x)))+(ggbtmpvarc),x))]))),ggbsort(normal([op(solve((((ggbtmpvara)*((x)^(2)))+((ggbtmpvarb)*(x)))+(ggbtmpvarc),x))])) ) ) )))
14<< {x=(1/2*(-ggbtmpvarb+√(ggbtmpvarb^2-4*ggbtmpvara*ggbtmpvarc))/ggbtmpvara),x=(1/2*(-ggbtmpvarb-√(ggbtmpvarb^2-4*ggbtmpvara*ggbtmpvarc))/ggbtmpvara)}
15>> caseval(regroup([[[ggbtanarg0:=ggbtmpvara],[ggbtanarg1:=sqrt((1)-((x)^(2)))],[ggbtanvar:=when(size(lname(ggbtanarg1) intersect [x]) == 0,lname(ggbtanarg1)[0],x)]],when((ggbtmpvara)[0]=='pnt',when((ggbtanarg1)[0]=='=',equation(tangent(ggbtanarg1,ggbtanarg0)),y=normal(subst(diff(ggbtanarg1,ggbtanvar),ggbtanvar=real(ggbtanarg0[1]))*(x-real(ggbtanarg0[1]))+subst(ggbtanarg1,ggbtanvar=real(ggbtmpvara[1])))),y=normal(subst(diff(ggbtanarg1,ggbtanvar),ggbtanvar=ggbtanarg0)*(x-(ggbtanarg0))+subst(ggbtanarg1,ggbtanvar=ggbtanarg0)))][1]))
15<< y=(-(ggbtmpvara*x-1)/√(-ggbtmpvara^2+1))
